x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sample/SampleIterator.java

```diff
@@ -174,8 +178,16 @@ private void finalStep(ActionListener<Payload> listener) {
                 if (docGroupsCounter == maxCriteria) {
                     List<SearchHit> match = matchSample(sample, maxCriteria);
                     if (match != null) {
-                        finalSamples.add(match);
-                        samples.add(new Sample(sampleKeys.get(responseIndex / maxCriteria), match));
+                        if (samplesDiscarded < limit.offset()) {
```

Comment:
WHy do you use both offset and limit?

Reply:
As long as we have Limit (see previous comment), IMHO it makes sense to have a complete support for it.
The logic is trivial and the added value could be significant: with a small additional effort on syntax, we'll have at least a basic support for pagination.

Reply:
Then why not having Limit as a node in the logical plan tree (vs a property of Sample plan)? (like SQL and EQL do it)

Reply:
Good point, checking

Reply:
Pagination won't work without state. 
Without reading the comments first, I also found this code confusing and had to check if we provide a non-null offset anywhere. I guess this could be clarified with a comment, but not sure where such a comment should be placed (if we ever add an offset, a comment placed here will nearly certainly not be updated/removed).
IMO, since the logic is trivial, it could be added when actually needed. But I guess it could also be kept.